pandas/core/arrays/datetimes.py

```diff
@@ -302,12 +284,10 @@ def _generate_range(cls, start, end, periods, freq, tz=None,
                         start = start.tz_localize(tz).asm8
                     if end is not None:
                         end = end.tz_localize(tz).asm8
-            else:
-                # Create a linearly spaced date_range in local time
-                start = start.tz_localize(tz)
-                end = end.tz_localize(tz)
-                arr = np.linspace(start.value, end.value, periods)
-                index = cls._simple_new(arr.astype('M8[ns]'), freq=None, tz=tz)
+        else:
+            # Create a linearly spaced date_range in local time
+            arr = np.linspace(start.value, end.value, periods)
```

Comment:
can do `dtype='M8[ns]'` in the linspace call

Reply:
can you do this (does it work?)

Reply:
This failed on the numpy compat build unfortunately.